cucumber/authenticators_jwt/features/authn_jwt_validate_restrictions.feature

```diff
@@ -0,0 +1,245 @@
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00099E Role must have at least one relevant annotation
+    """
+
+  Scenario: General annotation and invalid service specific annotation is given, 401 Error
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    Given I extend the policy with:
+    """
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/project-id: myproject
+        authn-jwt/aud: myaud
+        authn-jwt/raw/project-id: myproject
+        authn-jwt/raw/aud: wrong-aud
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/users
+      member: !host myapp
+    """
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "project-id": "myproject",
+      "aud": "myaud"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction 'aud' does not match with the corresponding value in the request
+    """
+
+  Scenario: Host without annotations, 401 Error
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    Given I extend the policy with:
+    """
+    - !host
+      id: myapp
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/users
+      member: !host myapp
+    """
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "project-id": "myproject",
+      "aud": "myaud"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00099E Role must have at least one relevant annotation
+    """
+
+  Scenario: Validate multiple annotations with incorrect values and 1 valid, 401 Error
+    And I successfully set authn-jwt "token-app-property" variable to value "host"
+    Given I extend the policy with:
+    """
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/raw/sub: invalid
+        authn-jwt/raw/project-path: invalid
+        authn-jwt/raw/project-id: myproject
+        authn-jwt/raw/aud: invalid
+
+    - !grant
+      role: !group conjur/authn-jwt/raw/users
+      member: !host myapp
+    """
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "sub": "sub",
+      "project-path":"path",
+      "project-id": "myproject",
+      "aud": "myaud"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00049E Resource restriction
+    """
+
+  Scenario: Validate multiple annotations with incorrect values and 1 valid, 401 Error
```

Comment:
isnt it the same as the test in line 154?

Reply:
Good catch thanks